### ISSUE TITLE
problems with releases API

### ISSUE THREAD

**Issue by targos** [philschatz/octokat.js#25]
I think I found two problems related to the releases API:

* `repo.releases(id).fetch()` works but logs an error: `BUG: Invalid Path. If this is actually a valid path then please update the URL_VALIDATOR. path=/repos/cheminfo-js/test/releases/1016794`
* `repo.releases.latest` does not exist

Also I do not understand how to use the release object to upload an asset.
```js
var repo = octo.repos('cheminfo-js', 'test');
var release = yield repo.releases(1016794).fetch() // OK with first logged bug
var contents = yield fs.readFile('./test.js');
yield release.upload('test.js', contents) // what should I do here ?
```

**Comment by philschatz** [comment]
Thanks for catching this!

On `#master` I added the additional routes for releases. Can you test the retrieval of releases? I don't normally use the releases API so cannot test it. 

Also, Regarding how to upload: I (incorrectly) assumed all of the `*_url` JSON fields were always GET requests. 

The ability to upload will probably require some additional work. I started it in the `#releases-upload` branch if you'd like to test or fix up.

to test uploading, you would need to do the following:

```js
yield release.upload('test.js', 'application/javascript', contents)
```

**Comment by philschatz** [comment]
The latest `octokat#0.4.8` on `npm` should have this feature in it. Thanks again!

**Comment by targos** [comment]
Hello and thank you for the fast work!
I am sorry I didn't have time to test it before your release.

The upload method seems to work correctly only for one asset. Then if I try to upload another one (different name and content), There is an error : `[Error: {"message":"Validation Failed","request_id":"13af8fb4-c7c4-11e4-8ddd-4cef8844458c","documentation_url":"https://developer.github.com/v3","errors":[{"resource":"ReleaseAsset","code":"already_exists","field":"name"}]}]`

My code:
```js
var buildFiles = yield build();
var releaseInfo = yield repo.releases.create({
    tag_name: `v${version}`,
    name: releaseMessage,
    prerelease: this.query.bump.startsWith('pre')
});
for (var i = 0; i < buildFiles.length; i++) {
        var file = yield fs.readFile(buildFiles[i].path);
        try {
            yield releaseInfo.upload(buildFiles[i].name, 'application/javascript', file);
        } catch (e) {
            console.error(e);
            console.error(`could not send ${buildFiles[i].path}`);
        }
    }
```

**Comment by philschatz** [comment]
Given the `{"resource":"ReleaseAsset","code":"already_exists","field":"name"}` part of the error it seems you might be trying to add more than one asset with the same name which it seems GitHub does not allow. Can you print out the list of names in `buildFiles`?

**Comment by targos** [comment]
of course, here is the log:
```
  grm:repo creating release +1ms
  grm:repo uploading 3 files: +436ms [ { name: 'test.js',
    path: '/home/mzasso/test/data/cheminfo-js/test/dist/test.js' },
  { name: 'test.min.js',
    path: '/home/mzasso/test/data/cheminfo-js/test/dist/test.min.js' },
  { name: 'test.min.js.map',
    path: '/home/mzasso/test/data/cheminfo-js/test/dist/test.min.js.map' } ]
  grm:repo uploading file 0 with name test.js +1ms
  grm:repo uploading file 1 with name test.min.js +891ms
{ message: 'Validation Failed',
  request_id: '17fa0c1e-c7fe-11e4-860b-9c575802439b',
  documentation_url: 'https://developer.github.com/v3',
  errors: 
   [ { resource: 'ReleaseAsset',
       code: 'already_exists',
       field: 'name' } ] }
could not send test.min.js
  grm:repo uploading file 2 with name test.min.js.map +403ms
{ message: 'Validation Failed',
  request_id: '1837558c-c7fe-11e4-84bb-8f2258d69b80',
  documentation_url: 'https://developer.github.com/v3',
  errors: 
   [ { resource: 'ReleaseAsset',
       code: 'already_exists',
       field: 'name' } ] }
could not send test.min.js.map

```

**Comment by targos** [comment]
I have a proposed change in #27.
Problematic line is https://github.com/philschatz/octokat.js/blob/master/src/replacer.coffee#L89
`value` belongs to the scope of the parent function so it is changed during the first upload.

**Comment by philschatz** [comment]
published as `0.4.9`; Thanks again!
